boards/remote/include/board.h

```diff
@@ -25,36 +24,81 @@
 
 #include "cpu.h"
 #include "periph/gpio.h"
+#include "periph/spi.h"
 
 #ifdef __cplusplus
  extern "C" {
 #endif
 
 /**
- * @brief   LED pin definitions and handlers
+ * @name LED pin definitions
  * @{
  */
-#define LED0_PIN            GPIO_PIN(3, 2)
-#define LED1_PIN            GPIO_PIN(3, 5)
-#define LED2_PIN            GPIO_PIN(2, 3)
-
-#define LED0_MASK           (1 << 2)
-#define LED1_MASK           (1 << 5)
-#define LED2_MASK           (1 << 3)
-
-#define LED0_ON             (GPIO_D->DATA &= ~LED0_MASK)
-#define LED0_OFF            (GPIO_D->DATA |=  LED0_MASK)
-#define LED0_TOGGLE         (GPIO_D->DATA ^=  LED0_MASK)
-
-#define LED1_ON             (GPIO_D->DATA &= ~LED1_MASK)
-#define LED1_OFF            (GPIO_D->DATA |=  LED1_MASK)
-#define LED1_TOGGLE         (GPIO_D->DATA ^=  LED1_MASK)
-
-#define LED2_ON             (GPIO_C->DATA &= ~LED2_MASK)
-#define LED2_OFF            (GPIO_C->DATA |=  LED2_MASK)
-#define LED2_TOGGLE         (GPIO_C->DATA ^=  LED2_MASK)
+#define LED_RED_GPIO        GPIO_PD5
+#define LED_GREEN_GPIO      GPIO_PD4
+#define LED_BLUE_GPIO       GPIO_PD3
 /** @} */
+/**
+ * @name Macros for controlling the on-board LEDs
+ * @{
+ */
+#define LED_RED_ON         gpio_set(LED_RED_GPIO)
+#define LED_RED_OFF        gpio_clear(LED_RED_GPIO)
+#define LED_RED_TOGGLE     gpio_toggle(LED_RED_GPIO)
+#define LED_GREEN_ON       gpio_set(LED_GREEN_GPIO)
+#define LED_GREEN_OFF      gpio_clear(LED_GREEN_GPIO)
+#define LED_GREEN_TOGGLE   gpio_toggle(LED_GREEN_GPIO)
+#define LED_BLUE_ON        gpio_set(LED_BLUE_GPIO)
+#define LED_BLUE_OFF       gpio_clear(LED_BLUE_GPIO)
+#define LED_BLUE_TOGGLE    gpio_toggle(LED_BLUE_GPIO)
 
+#define LED_ALL_OFF        LED_GREEN_OFF; \
+                           LED_RED_OFF;   \
+                           LED_BLUE_OFF
+/* Output is color white */
+#define LED_ALL_ON         LED_BLUE_ON;   \
+                           LED_RED_ON;    \
+                           LED_GREEN_ON
+#define LED_YELLOW_ON      LED_BLUE_OFF;  \
+                           LED_GREEN_ON;  \
+                           LED_RED_ON
+#define LED_YELLOW_OFF     LED_GREEN_OFF; \
+                           LED_RED_OFF
+#define LED_YELLOW_TOGGLE  LED_GREEN_TOGGLE; \
+                           LED_RED_TOGGLE
+#define LED_PURPLE_ON      LED_GREEN_OFF;    \
+                           LED_BLUE_ON;      \
+                           LED_RED_ON
+#define LED_PURPLE_OFF     LED_BLUE_OFF;     \
+                           LED_RED_OFF
+#define LED_PURPLE_TOGGLE  LED_BLUE_TOGGLE;  \
+                           LED_RED_TOGGLE
+/* Take LED_COLOR as argument, i.e LED_RED */
+#define LED_FADE(led)                         \
```

Comment:
If you want to do the fancy LED fading stuff, please put this into a library outside of the board code and include it.